crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,441 @@
+//! Components used to control the layout of UI-[`Node`](super::Node) entities.
+use crate::{Size, UiRect, Val};
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::prelude::Component;
+use bevy_reflect::prelude::*;
+use serde::{Deserialize, Serialize};
+
+use self::flex::FlexLayout;
+
+/// Controls which layout algorithm is used to position this UI node
+#[derive(Component, Copy, Clone, PartialEq, Debug, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum Layout {
+    /// Use the absolute position and size specified
+    PositionLayout,
+    /// Use the [Flexbox](https://cssreference.io/flexbox/) layout algorithm
+    ///
+    /// As implemented by [`taffy`]: some bugs or limitations may exist;
+    /// [please file an issue!](https://github.com/DioxusLabs/taffy/issues)
+    FlexLayout(FlexLayout),
+}
+
+impl Default for Layout {
+    fn default() -> Self {
+        Layout::FlexLayout(FlexLayout::default())
+    }
+}
+
+/// The strategy used to position this node
+#[derive(
+    Component, Copy, Clone, PartialEq, Eq, Debug, Default, Serialize, Deserialize, Reflect,
+)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum PositionType {
+    /// Relative to all other nodes with the [`PositionType::Relative`] value
+    #[default]
+    Relative,
+    /// Positioned in relation to its parent,
+    /// but calculated as if this was the only child of its parent.
+    Absolute,
+}
+
+/// The offset of a UI node from its base position
+///
+/// Layout is performed according to the [`Layout`], and then this value is added at the end.
+/// When this is [`Layout::PositionLayout`], this value will represent the absolute position of the UI node.
+/// To check the final position of a UI element, read its [`Transform`](bevy_transform::components::Transform) component.
+#[derive(
+    Component,
+    Deref,
+    DerefMut,
+    Copy,
+    Clone,
+    PartialEq,
+    Debug,
+    Default,
+    Serialize,
+    Deserialize,
+    Reflect,
+)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub struct Offset(pub UiRect);
```

Comment:
I'm not too sure about the rename of `Position -> Offset`. While I think it might be more accurate, most people will know this as `Position` from CSS or CSS-like frameworks. That is at least my own personal experience.

Reply:
I would prefer `Offset`: there are more than enough confusing historical CSS properties out there, and bevy doesn't need to follow them if a clearer name exists.

Reply:
Yeah I'm coming around on this one. I agree that Offset is clearer in what it does

Reply:
I disagree, I prefer position. This not only matches CSS, this is also a common pattern that position of a child is the offset from his parent. Offset itself however can mean anything if we don't specify relative to what.